HTMLSerializer.js

```diff
@@ -0,0 +1,326 @@
+/**
+ * HTML serializer that takes a document and synchronously stores it as an array
+ * of strings and stores enough state to later asynchronously convert it into an
+ * html text file.
```

Comment:
Because HTMLSerializer now contains fillSrcHoles, I think this comment should be changed to:

"HTML Serializer that takes a document and synchronously stores it as an array of strings, then asynchronously retrieves data URLs for same-origin images.  It stores enough state to later convert it into an html text file."

What do you think?